src/epublius/metadata.py

```diff
@@ -128,14 +128,19 @@ def get_chapter_title(self):
 
     def get_chapter_doi(self):
         '''
-        Retrieve chapter DOI based on the text of <p class=doi>
+        Retrieve chapter DOI based on the text of <p class="doi">
         (this contains both copyright statement and DOI link)
         '''
         # Not all chapters will have DOIs
         doi = None
 
         doi_node = self.soup.find('p', class_='doi')
```

Comment:
You can just do the initial search with `doi_node = soup.find('p', class_=lambda x: x and 'doi' in x.split())` (using `split()` to cover all cases), instead of searching twice